app/actions/deployment_create.rb

```diff
@@ -75,7 +75,13 @@ def create(app:, user_audit_info:, message:)
 
         deployment
       rescue RevisionResolver::NoUpdateRollback, Sequel::ValidationFailed, AppStart::InvalidApp => e
-        error = DeploymentCreate::Error.new(e.message)
+        space_quota_errors = [:space_quota_exceeded.to_s, :space_app_instance_limit_exceeded.to_s]
```

Comment:
I think we'd want to do something similar for org-level quotas: https://github.com/cloudfoundry/cloud_controller_ng/blob/12ebacb24ad9c33b5e69c5b7a0bbc79a80c25974/app/models/runtime/process_model.rb#L257